Source/WebKit/UIProcess/Launcher/glib/XDGDBusProxy.cpp

```diff
@@ -41,21 +41,16 @@
 
 namespace WebKit {
 
-CString XDGDBusProxy::makePath(const char* dbusAddress)
+const CString& XDGDBusProxy::dbusSessionPath()
 {
-    if (!dbusAddress || !g_str_has_prefix(dbusAddress, "unix:"))
-        return { };
-
-    if (const char* path = strstr(dbusAddress, "path=")) {
-        path += strlen("path=");
-        const char* pathEnd = path;
-        while (*pathEnd && *pathEnd != ',')
-            pathEnd++;
-
-        return CString(path, pathEnd - path);
-    }
+    static CString path { g_build_filename(g_get_user_runtime_dir(), BASE_DIRECTORY, "bus", nullptr) };
+    return path;
```

Comment:
> This is leaking, g_build_filename returns a new allocated string and CString copies the given string.

Oops. :S

> Why does this belong here? it's not using anything from the XDGDbusProxy, it's the path where the launcher wants to bind the socket, so it can be moved there using GUniquePtr instead of CString. 

The session bus path can be moved to BubblewrapLauncher.cpp, indeed. I like that idea. Except: it's logically paired with the accessibility bus path, and moving that would be more awkward due to the call to PlatformDisplay::setAccessibilityBusAddress. That call really wants to stay in XDGDBusProxy.cpp because we want to call it only once, when the accessibility proxy is actually launched. I suppose it would be harmless to call it repeatedly, but that's not quite as nice. I suppose we could have XDGDBusProxy::accessibilityProxy accept the path as a parameter, so that it can be passed to PlatformDisplay::setAccessibilityBusAddress that way. Sound good?

> I'm not sure it's worth caching it, since the dbus proxy is only launched once.

But the path is needed once per process launched, it's not actually needed when the proxy itself is launched. And the path never changes. Caching is not important though, just a very tiny optimization.